x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/sort/IpBucketedSort.java

```diff
@@ -77,10 +69,7 @@ public class IpBucketedSort implements Releasable {
     private ByteArray values;
 
     public IpBucketedSort(BigArrays bigArrays, SortOrder order, int bucketSize) {
-        this.bigArrays = bigArrays;
-        this.order = order;
-        this.bucketSize = bucketSize;
-        heapMode = new BitArray(0, bigArrays);
+        this.common = new BucketedSortCommon(bigArrays, order, bucketSize);
```

Comment:
No inheritance? Shouldn't `final` methods be safe to use?

Reply:
I sure could have inherited it. I started that way because it felt easier but the ctor with the sub-types and the closing and.... for that, at least, it felt easier to compose.